Client/Frontend/Browser/BrowserViewController/BrowserViewController+ProductNotification.swift

```diff
@@ -0,0 +1,323 @@
+        let checkDate = Preferences.ProductNotificationBenchmarks.ongoingEducationCheckDate.value
+        let isProductNotificationsValid = todayInSeconds <= checkDate
+        
+        var notificationShown = false
+        let contentBlockerStats = selectedTab.contentBlocker.stats
+
+        // Step 1: First Time Block Notification
+        if isProductNotificationsValid,
+           !Preferences.ProductNotificationBenchmarks.firstTimeBlockingShown.value,
+           contentBlockerStats.total > 0 {
+            
+            notifyFirstTimeBlock(theme: Theme.of(selectedTab))
+            
+            Preferences.ProductNotificationBenchmarks.firstTimeBlockingShown.value = true
+            Preferences.ProductNotificationBenchmarks.ongoingEducationCheckDate.value = Date().timeIntervalSince1970 + 7.days
+            
+            notificationShown = true
+        }
+        
+        // Step 2: Load a video on a streaming site
+        guard !notificationShown else { return }
+
+        let isVideoStreamingSite = (selectedTab.canonicalURL?.absoluteString.contains("youtube") == true
+                                        || selectedTab.canonicalURL?.absoluteString.contains("vimeo") == true)
+            
+        if isProductNotificationsValid,
+           !Preferences.ProductNotificationBenchmarks.videoAdBlockShown.value,
+           isVideoStreamingSite {
+            
+            notifyVideoAdsBlocked(theme: Theme.of(selectedTab))
+            notificationShown = true
+        }
+        
+        // Step 3: 20+ Trackers and Ads Blocked
+        guard !notificationShown else { return }
+
+        if isProductNotificationsValid,
+           !Preferences.ProductNotificationBenchmarks.privacyProtectionBlockShown.value,
+           contentBlockerStats.total > benchmarkNumberOfTrackers {
+            
+            notifyPrivacyProtectBlock(theme: Theme.of(selectedTab))
+            notificationShown = true
+        }
+        
+        // Step 4: Https Upgrade
+        guard !notificationShown else { return }
+
+        if isProductNotificationsValid,
+           !Preferences.ProductNotificationBenchmarks.httpsUpgradeShown.value,
+           contentBlockerStats.httpsCount > 0 {
+            
+            notifyHttpsUpgrade(theme: Theme.of(selectedTab))
+            notificationShown = true
+        }
+        
+        // Step 5: Share Brave Benchmark Tiers
+        guard !notificationShown else { return }
+
+        if !Preferences.ProductNotificationBenchmarks.allTiersShown.value {
+            let numOfTrackersAds = BraveGlobalShieldStats.shared.adblock + BraveGlobalShieldStats.shared.trackingProtection
+            let allBenchmarkTiers = BenchmarkTrackerCountTier.allCases
+            let savedTrackerTierCount = Preferences.ProductNotificationBenchmarks.trackerTierCount.value
+            
+            for (index, tier) in (allBenchmarkTiers.filter { numOfTrackersAds > $0.rawValue }).enumerated() {
```

Comment:
This loop is O(N^2), is it possible to use `for-where` loop? 

```swift
for (index, tier) in allBenchmarkTiers.enumerated() where numberOfTrackerAds > tier.rawValue {
    print(name)
}
```

Reply:
I will spend some time to make this logic more efficient.